presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -290,6 +291,16 @@ std::string SystemConfig::prestoVersion() const {
   return requiredProperty(std::string(kPrestoVersion));
 }
 
+std::string SystemConfig::poolType() const {
+    static const std::unordered_set<std::string> validTypes = {"LEAF", "INTERMEDIATE", "DEFAULT"};
+    auto value = requiredProperty(std::string(kPoolType));
```

Comment:
Should this be optionalProperty instead ? Configs which don't have pool-type will cause failures post this change. So this could be very disruptive. Could that be the reason for all the test failures ? I haven't looked at the details but the timeouts most likely could be because the workers didn't start up.

Reply:
i thought the default will always be initialized when we put this in the registeredProps_ map i can try with optional